cmd/sourced/compose/workdir/common.go

```diff
@@ -47,14 +50,43 @@ func (f *envFile) String() string {
 		gitbaseSiva = "true"
 	}
 
+	// limit CPU for containers
+	var gitbaseLimitCPU, gitcollectorLimitCPU string
+	dockerCPUs, err := dockerNumCPU()
+	if err != nil { // show warning
+		fmt.Println(err)
+	}
+	// apply gitbase resource limits only when docker runs without any global limits
+	// it's default behaviour on linux
+	if runtime.NumCPU() == dockerCPUs {
+		gitbaseLimitCPU = fmt.Sprintf("%.1f", float32(dockerCPUs)-0.1)
+	}
+	gitcollectorLimitCPU = fmt.Sprintf("%.1f", float32(dockerCPUs)/2-0.1)
+
+	// limit memory for containers
+	var gitbaseLimitMem string
+	dockerMem, err := dockerTotalMem()
+	if err != nil { // show warning
+		fmt.Println(err)
+	}
+	// apply memory limits only when only when docker runs without any global limits
+	// it's default behaviour on linux
+	if dockerMem == memory.TotalMemory() {
```

Comment:
is there any advantage to do this instead of `if runtime.GOOS == "linux"`?

Reply:
You can [limit mem/cpu on linux globally](https://medium.com/@asishrs/docker-limit-resource-utilization-using-cgroup-parent-72a646651f9d) or give all available resources to docker on macOs/Windows. That isn't a default behavior but a valid one.